File: src/main/java/com/github/hdzitao/editstarters/dependency/Bom.java
package com.github.hdzitao.editstarters.dependency;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class Bom implements Point {
    protected String groupId;
    protected String artifactId;
    protected String version;

    public Bom(String groupId, String artifactId, String version) {
        this.groupId = groupId;
        this.artifactId = artifactId;
        this.version = version;
    }

    public Bom(String groupId, String artifactId) {
        this(groupId, artifactId, null);
    }

    @Override
    public String point() {
        return this.groupId + ":" + this.artifactId;
    }
}

File: src/main/kotlin/io/github/hdzitao/editstarters/dependency/Bom.kt
package io.github.hdzitao.editstarters.dependency

/**
 * bom
 *
 * @version 3.2.0
 */
open class Bom(
    var groupId: String? = null,
    var artifactId: String? = null,
    var version: String? = null,
) : Point {

    override fun point() = "$groupId:$artifactId"
}